Index: /trunk/railties/lib/rails_generator/lookup.rb
===================================================================
--- /trunk/railties/lib/rails_generator/lookup.rb (revision 4310)
+++ /trunk/railties/lib/rails_generator/lookup.rb (revision 6101)
@@ -101,5 +101,5 @@
             sources << PathSource.new(:lib, "#{::RAILS_ROOT}/lib/generators")
             sources << PathSource.new(:vendor, "#{::RAILS_ROOT}/vendor/generators")
-            sources << PathSource.new(:plugins, "#{::RAILS_ROOT}/vendor/plugins/**/generators")
+            sources << PathSource.new(:plugins, "#{::RAILS_ROOT}/vendor/plugins/*/**/generators")
           end
           sources << PathSource.new(:user, "#{Dir.user_home}/.rails/generators")
Index: /trunk/railties/lib/tasks/rails.rb
===================================================================
--- /trunk/railties/lib/tasks/rails.rb (revision 5469)
+++ /trunk/railties/lib/tasks/rails.rb (revision 6101)
@@ -6,3 +6,3 @@
 # Load any custom rakefile extensions
 Dir["#{RAILS_ROOT}/lib/tasks/**/*.rake"].sort.each { |ext| load ext }
-Dir["#{RAILS_ROOT}/vendor/plugins/**/tasks/**/*.rake"].sort.each { |ext| load ext }
+Dir["#{RAILS_ROOT}/vendor/plugins/*/**/tasks/**/*.rake"].sort.each { |ext| load ext }
Index: /trunk/railties/lib/tasks/testing.rake
===================================================================
--- /trunk/railties/lib/tasks/testing.rake (revision 5263)
+++ /trunk/railties/lib/tasks/testing.rake (revision 6101)
@@ -109,9 +109,9 @@
       t.pattern = "vendor/plugins/#{ENV['PLUGIN']}/test/**/*_test.rb"
     else
-      t.pattern = 'vendor/plugins/**/test/**/*_test.rb'
+      t.pattern = 'vendor/plugins/*/**/test/**/*_test.rb'
     end
 
     t.verbose = true
   end
-  Rake::Task['test:plugins'].comment = "Run the plugin tests in vendor/plugins/**/test (or specify with PLUGIN=name)"
+  Rake::Task['test:plugins'].comment = "Run the plugin tests in vendor/plugins/*/**/test (or specify with PLUGIN=name)"
 end
Index: /trunk/railties/CHANGELOG
===================================================================
--- /trunk/railties/CHANGELOG (revision 6069)
+++ /trunk/railties/CHANGELOG (revision 6101)
@@ -1,3 +1,5 @@
 *SVN*
+
+* Plugins may be symlinked in vendor/plugins.  #4245 [brandon, progrium@gmail.com]
 
 * Resource generator depends on the model generator rather than duplicating it.  #7269 [bscofield]
